src/logging.ts

```diff
@@ -2,5 +2,7 @@ import Logger from 'bunyan';
 
 export type LoggerInterface = Pick<
   Logger,
-  'child' | 'trace' | 'debug' | 'info' | 'warn' | 'error' | 'fatal'
->;
+  'trace' | 'debug' | 'info' | 'warn' | 'error' | 'fatal'
+> & {
+  child: (...args: Parameters<Logger['child']>) => LoggerInterface;
```

Comment:
This is the main change. `child()` needs to return the interface rather than a concrete type to be backwards compatible.